File: tyto/csharp/core/Tyto.Utils/OsUtil/OsUtil.cs
namespace Tyto.Utils;

using System;

public static class OsUtil
{
	/// <summary>
	/// 获取软件的位数
	/// </summary>
	public static int GetBit()
	{
		return IntPtr.Size * 8;
	}

	/// <summary>
	/// 是否为64位软件
	/// </summary>
	public static bool Is64Bit()
	{
		return GetBit() == 64;
	}

	/// <summary>
	/// 是否为32位软件
	/// </summary>
	public static bool Is32Bit()
	{
		return GetBit() == 32;
	}

	/// <summary>
	/// 是否为Windows操作系统
	/// </summary>
	public static bool IsWindows()
	{
		return Environment.OSVersion.Platform == PlatformID.Win32NT;
	}

	/// <summary>
	/// 是否为Linux操作系统
	/// </summary>
	public static bool IsLinux()
	{
		return Environment.OSVersion.Platform == PlatformID.Unix;
	}

	/// <summary>
	/// 是否为MacOSX操作系统
	/// </summary>
	public static bool IsMacOsx()
	{
		return Environment.OSVersion.Platform == PlatformID.MacOSX;
	}
}

File: tyto/go/core/osutil/osutil.go
package osutil

import (
	"runtime"
	"unsafe"
)

// 获取软件位数
func GetBit() int32 {
	return int32(unsafe.Sizeof(uintptr(0)) * 8)
}

// 是否64位软件
func Is64Bit() bool {
	return GetBit() == 64
}

// 是否32为软件
func Is32Bit() bool {
	return GetBit() == 32
}

// 是否windows系统
func IsWindows() bool {
	return runtime.GOOS == "windows"
}

// 是否linux系统
func IsLinux() bool {
	return runtime.GOOS == "linux"
}

// 是否mac osx系统
func IsMacOsx() bool {
	return runtime.GOOS == "darwin"
}
